lib/http3/server.c

```diff
@@ -1087,7 +1093,7 @@ static void handle_buffered_input(struct st_h2o_http3_server_stream_t *stream, i
                                 stream->req.req_body_bytes_received < stream->req.content_length
                                     ? H2O_HTTP3_ERROR_REQUEST_INCOMPLETE
                                     : H2O_HTTP3_ERROR_GENERAL_PROTOCOL,
-                                in_generator);
+                                in_generator, 1);
```

Comment:
I think we should continue to RST_STREAM if the body length and content-length header are not consistent with each other.

Reply:
Thank you for the review.

When the request is broken, we can certainly continue the current behavior of converting an early response to resets.

I think I actually like the idea of changing the behavior in the cases only when the generator (i.e., the HTTP client connected to the back end server) signals the error.

I've adopted that approach in beff704, and we only have three cases that set the `reset_only_if_open` flag! WDYT?